python/triton/compiler/code_generator.py

```diff
@@ -1052,19 +1052,18 @@ def _verify_loop_carried_variable(self, name, loop_val, live_val):
         assert _is_triton_value(loop_val), f'cannot reassign constexpr {name} in the loop'
         assert _is_triton_value(live_val), f'cannot reassign constexpr {name} in the loop'
         assert type(loop_val) is type(live_val), f'Loop carried variable {name} changed type'
-        if hasattr(loop_val, 'type'):
-            assert loop_val.type == live_val.type, \
-                f'Loop-carried variable "{name}" has initial type {live_val.type} '\
-                f'but is re-assigned to {loop_val.type} in loop! '\
-                f'Please make sure that the type stays consistent.'
+        assert not _is_triton_tensor(loop_val) or loop_val.type == live_val.type, \
```

Comment:
`_is_triton_tensor` doesn't include `buffered_tensor` type. Why rolling back to `_is_triton_tensor` ?